mentoring/answer.py

```diff
@@ -24,39 +21,146 @@
+            name = self.name
+        # Consistency check - we should have a name by now
+        if not name:
+            raise ValueError('AnswerBlock.name field need to be set to a non-null/empty value')
+
+        student_id = self._get_student_id()
+        course_id = self._get_course_id()
+
+        answer_data, _ = Answer.objects.get_or_create(
+            student_id=student_id,
+            course_id=course_id,
+            name=name,
+        )
+        return answer_data
+
+    def validate_field_data(self, validation, data):
+        """
+        Validate this block's field data.
+        """
+        super(AnswerMixin, self).validate_field_data(validation, data)
+
+        def add_error(msg):
+            validation.add(ValidationMessage(ValidationMessage.ERROR, msg))
+
+        if not data.name:
+            add_error(u"A Question ID is required.")
+
+    def _(self, text):
+        """ translate text """
+        return self.runtime.service(self, "i18n").ugettext(text)
+
+
+@XBlock.needs("i18n")
+class AnswerBlock(AnswerMixin, StepMixin, StudioEditableXBlockMixin, XBlock):
     """
     A field where the student enters an answer
 
     Must be included as a child of a mentoring block. Answers are persisted as django model instances
     to make them searchable and referenceable across xblocks.
     """
-    read_only = Boolean(help="Display as a read-only field", default=False, scope=Scope.content)
-    default_from = String(help="If specified, the name of the answer to get the default value from",
-                          default=None, scope=Scope.content)
+    name = String(
+        display_name=_("Question ID (name)"),
+        help=_("The ID of this block. Should be unique unless you want the answer to be used in multiple places."),
+        default="",
+        scope=Scope.content
+    )
+    default_from = String(
+        display_name=_("Default From"),
+        help=_("If a question ID is specified, get the default value from this answer."),
+        default=None,
+        scope=Scope.content
+    )
+    min_characters = Integer(
+        display_name=_("Min. Allowed Characters"),
+        help=_("Minimum number of characters allowed for the answer"),
+        default=0,
+        scope=Scope.content
+    )
+    question = String(
+        display_name=_("Question"),
+        help=_("Question to ask the student"),
+        scope=Scope.content,
+        default=""
+    )
+    weight = Float(
+        display_name=_("Weight"),
+        help=_("Defines the maximum total grade of the answer block."),
+        default=1,
+        scope=Scope.settings,
+        enforce_type=True
+    )
+
+    editable_fields = ('question', 'name', 'min_characters', 'weight', 'default_from')
+
+    @property
+    def studio_display_name(self):
+        if not self.lonely_step:
+            return self._(u"Question {number}").format(number=self.step_number)
+        return self._(u"Question")
+
+    def __getattribute__(self, name):
```

Comment:
Another option, if you're only overriding `display_name`, would be to add your own python-descriptor for `display_name` that is read-only and proxies to `studio_display_name`.

I'm a bit confused, though, around your statement of `display_name_with_default`, and about not wanting to add a `display_name` field to the class. What outcome are you trying to get? What are you trying to avoid?